sul-dlss/course_reserves
# Set up rake task to get course information from MaIS APIs

What this pull request does:

- Adds a rake task to use the MaIS course term and individual course APIs to retrieve course information for the current term and the next term.  

What this pull request does not do:
- This pull request does not replace the fetch_xml rake task.  This means the app will still use the XML from registry harvester until a future pull request ties the JSON resulting from the rake task to what is read into the app for displaying the course list.
- This pull request does not add fetch_courses to schedule.rb.  A future pull request will make that change.  

How to test this pull request:
- After putting the sul-harvester.key and sul-harvester.cert files in the config directory, run "rake fetch_courses" from the main directory.    The process takes a few hours but should result in two json files in the /course_work_xml directory. 

More details:
The fetch_courses rake task uses a new class called CourseAPI class to parse the course term info returned from the MaIS Course Term API.  CourseAPI also retrieves section and instructor information, and generates the JSON that the fetch_courses rake task writes to the course_work_xml directory. The process of generating these JSON files takes a few hours, so this step is better left as a rake task that can run independently and then make the info available to the app. One JSON file per term is generated. fetch_xml requested the current and next immediate term, so that pattern remains the same.
